reana_workflow_controller/app.py

```diff
@@ -42,75 +42,51 @@
-        # be checked but since this is a proof of
-        # concept we suppose that it exists.
-        for pos, line in enumerate(lines[3:]):
-            lines[pos+3] = lines[pos+3].strip()
-            tmp_nums = []
-            for num in line.split(','):
-                tmp_nums.append(int(num))
-            nums.append(tmp_nums)
-    else:
-        tmp_nums = []
-        for num in input_file.split(','):
-            tmp_nums.append(int(num))
-        nums.append(tmp_nums)
-
-    return lines[1].strip(), lines[2].strip(), '\n'.join(lines[3:])
-
-
-@app.route('/fibonacci', methods=['GET', 'POST'])
-def fibonacci_endpoint():
-    if request.method == 'GET':
-        return render_template('bg-form.html')
-
-    # Calculate fibonacci in background
-    if request.method == 'POST':
-        try:
-            if request.json:
-                task_weight = request.json['weight']
-                queue = experiment_to_queue[request.json['experiment']]
-                input_file = base64.decodestring(request.json['input-file'])
-                docker_img, cmd, fib_file = check_fibonacci_workflow(
-                    input_file
-                )
+@app.route('/yadage', methods=['POST'])
+def yadage_endpoint():
+    """Create a new job.
 
-                fibonacci.apply_async(
-                    args=[docker_img, cmd, task_weight, fib_file,
-                          request.json['experiment']],
-                    queue='fibo-{}'.format(queue)
-                )
-                return 'Workflow successfully launched'
-            else:
-                task_weight = request.form['weight']
-                queue = experiment_to_queue[request.form['experiment']]
-                input_file = request.form['input-file']
-                docker_img, cmd, fib_file = check_fibonacci_workflow(
-                    input_file
-                )
+    .. http:post:: /yadage
 
-                fibonacci.apply_async(
-                    args=[docker_img, cmd, task_weight, fib_file,
-                          request.form['experiment']],
-                    queue='fibo-{}'.format(queue)
-                )
+        This resource is expecting JSON data with all the necessary
+        information to run a yadage workflow.
 
-                flash('Workflow successfully launched')
-                return redirect(url_for('fibonacci_endpoint'))
-        except (KeyError, ValueError):
-            traceback.print_exc()
-            abort(400)
+        **Request**:
 
+        .. sourcecode:: http
 
-@app.route('/yadage', methods=['GET', 'POST'])
-def yadage_endpoint():
+            POST /yadage HTTP/1.1
+            Content-Type: application/json
+            Host: localhost:5000
+
+            {
+                "experiment": "atlas",
+                "toplevel": "from-github/testing/scriptflow",
+                "workflow": "workflow.yml",
+                "nparallel": "100",
+                "preset_pars": FIXME
```

Comment:
@lukasheinrich could you please provide an example value for this field? :smile: